bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -291,13 +291,24 @@ public void clearHomekitPairings() {
     }
 
     @Override
-    public void onChanged(final List<CidrAddress> list, final List<CidrAddress> list1) {
-        logger.trace("restarting homekit bridge on network interface changes.");
-        stopHomekitServer();
-        try {
-            startHomekitServer();
-        } catch (IOException e) {
-            logger.warn("could not initialize HomeKit bridge: {}", e.getMessage());
+    public void onChanged(final List<CidrAddress> added, final List<CidrAddress> removed) {
+        logger.trace("restarting HomeKit bridge on network interface changes.");
+        removed.forEach(i -> {
+            logger.trace("removed interface {}", i.getAddress().toString());
+            if (i.getAddress().equals(networkInterface)) {
+                this.bridge.stop();
+                this.homekitServer.stop();
```

Comment:
No need to check that this.bridge and this.homekitServer are not null ?

Reply:
valid point. fixed